stream_alert/rule_processor/rules_engine.py

```diff
@@ -497,32 +336,34 @@ def rule_analysis(self, record, rule, payload, alerts):
             alerts (list): The current list of Alert instances.
                 If the rule returns True on the record, a new Alert instance is added to this list.
         """
-        rule_result = StreamRules.process_rule(record, rule)
-        if rule_result:
-            # when threat intel enabled, normalized records will be re-analyzed by
-            # all rules. Thus we need to check duplication.
-            if self._threat_intel and self.check_alerts_duplication(record, rule, alerts):
-                return
-
-            # Combine the required alert outputs with the ones for this rule
-            all_outputs = self._required_outputs_set.union(set(rule.outputs or []))
-
-            alert = Alert(
-                rule.rule_name, record, all_outputs,
-                cluster=os.environ['CLUSTER'],
-                context=rule.context,
-                log_source=str(payload.log_source),
-                log_type=payload.type,
-                rule_description=rule.rule_function.__doc__,
-                source_entity=payload.entity,
-                source_service=payload.service()
-            )
-
-            LOGGER.info('Rule [%s] triggered alert [%s] on log type [%s] from entity \'%s\' '
-                        'in service \'%s\'', rule.rule_name, alert.alert_id, payload.log_source,
-                        payload.entity, payload.service())
-
-            alerts.append(alert)
+        rule_result = rule.process(record)
+        if not rule_result:
+            return
+
+        # when threat intel enabled, normalized records will be re-analyzed by
+        # all rules. Thus we need to check duplication.
+        if self._threat_intel and self.check_alerts_duplication(record, rule, alerts):
+            return
+
+        # Combine the required alert outputs with the ones for this rule
+        all_outputs = self._required_outputs_set.union(set(rule.outputs or []))
```

Comment:
The `set(rule.outputs or [])` logic feels like it should be in the `Rule` class, because we may need it elsewhere.  Consider something like:

```python
class Rule:
...
    @property
    def output_set(self):
        return set(self.outputs or [])
```
Or else setting `rule.outputs` to a set during Rule instantiation